explainaboard/visualizers/draw_charts.py

```diff
@@ -0,0 +1,265 @@
+            y_errs = [
+                (
+                    [x.value - unwrap(x.confidence_score_low) for x in y],
+                    [unwrap(x.confidence_score_high) - x.value for x in y],
+                )
+                for y in overall_results
+            ]
+
+        make_bar_chart(
+            ys,
+            output_dir,
+            'overall',
+            output_fig_format='png',
+            fig_size=(8, 6),
+            sys_names=sys_names,
+            errs=y_errs,
+            title=None,
+            xticklabels=overall_metric_names,
+            ylabel='metric value',
+        )
+
+    # --- analysis results
+    analysis_results: list[list[AnalysisResult]] = [
+        unwrap(x.results.analyses) for x in report_info
+    ]
+    if any([len(x) != len(analysis_results[0]) for x in analysis_results]):
+        raise ValueError(
+            f'mismatched number of analyses: {[len(x) for x in analysis_results]}'
+        )
+
+    # Bucket performance: feature name, for example, sentence length
+    for analysis_result in progress(zip(*analysis_results)):
+        if any([x.name != analysis_result[0].name for x in analysis_result]):
+            raise ValueError(
+                f'mismatched analyses: {[x.name for x in analysis_result]}'
+            )
+
+        if all([isinstance(x, BucketAnalysisResult) for x in analysis_result]):
+            plot_buckets(analysis_result, output_dir, sys_names)
+        elif all([isinstance(x, ComboCountAnalysisResult) for x in analysis_result]):
+            plot_combo_counts(analysis_result, output_dir, sys_names)
+        else:
+            raise ValueError('illegal types of analyses')
+
+
+def main():
+
+    parser = argparse.ArgumentParser(
+        description='Draw Histogram for ExplainaBoard Report'
+    )
+
+    parser.add_argument('--task', type=str, required=False, help="the task name")
+
+    parser.add_argument(
+        '--reports',
+        type=str,
+        required=True,
+        nargs="+",
+        help="the directories of reports. Multiple one should be separated by space, "
+        "for example: report1 report2",
+    )
+
+    parser.add_argument(
+        '--sys_names',
+        type=str,
+        required=False,
+        nargs="+",
+        default=None,
+        help="names of each system, separated by spaces",
+    )
+
+    parser.add_argument(
+        '--output_dir',
+        type=str,
+        required=False,
+        default="figures",
+        help="the directory of generated figures",
+    )
+
+    args = parser.parse_args()
+
+    reports = args.reports
+    sys_names = args.sys_names
+    output_dir = "figures" if args.output_dir is None else args.output_dir
+
+    draw_charts_from_reports(reports, output_dir, sys_names=sys_names)
+
+
+if __name__ == '__main__':
+    main()
+    # explainaboard --system_outputs ./data/system_outputs/multilingual/json/
+    # CL-mlpp15out1sum/marc/test-de_9330.json
+    # python draw_charts.py --reports ../log.res
```

Comment:
It looks this is the example of running this script, which should be put on the top of the file.

Reply:
Done, I also made it more verbose.